src/flb_metrics.c

```diff
@@ -324,8 +324,7 @@ int flb_metrics_fluentbit_add(struct flb_config *ctx, struct cmt *cmt)
     /* get hostname */
     ret = gethostname(hostname, sizeof(hostname) - 1);
     if (ret == -1) {
-        strncpy(hostname, "unknown", 7);
-        hostname[7] = '\0';
+        strcpy(hostname, "unknown");
```

Comment:
hostname will need the ending NULL byte

Reply:
Unless I'm wrong, the string literal will contain a NULL byte (https://stackoverflow.com/questions/17943411/) that will be copied by `strcpy`.

Reply:
you are right, thanks